MP-5131 Table Feature Batch actions (#243)
spryker/spryker-docs: docs/marketplace/dev/front-end/table-design/table-features/table-feature-batch-actions.md

---
title: Table Feature Batch Actions
description: This article provides details about the Table Feature Batch Actions component in the Components Library.
template: concept-topic-template
---

This article provides details about the Table Feature Batch Actions component in the Components Library.

## Overview

Table Feature Batch Actions is a feature of the Table Component that allows triggering batch/multiple actions from rows.
As Table Feature Batch Actions based on the [Table Feature Selectable](/docs/marketplace/dev/front-end/table-design/table-features/table-feature-selectable.html), 
make sure to register and to enable this feature via table config.
Batch actions are functions that may be performed on multiple items within a table. 

> Once the user selects at least one row from the table, the batch action bar appears at the top of the table, 
> presenting the user with actions they can take. 

Reviewer: ```suggestion
As soon as at least one row is selected in the table, the batch action bar with allowed actions will appear at the top of the table.
```

> To exit or escape `batch action mode`, the user can deselect the items.
> See an example below, how to use the Batch Actions feature.

Reviewer: ```suggestion
To escape the `batch action mode`, it is necessary to unselect the table rows.

See an example below, how to use the Batch Actions feature.
```

Feature Configuration:

`enabled` - will enable feature via config.  
`noActionsMessage` - error message text.  
`actions` - an array with actions that will be displayed in the top bar and their type of 
registered Action (see more about [Actions](/docs/marketplace/dev/front-end/ui-components-library/actions/)).   
`rowIdPath` - gets row `id` via column `id` (in the example below `Sku` column).  
`availableActionsPath` - path to an array with available action IDs in the top bar (supports nested 
objects using dot notation for ex. `prop.nestedProp`).   

```html
<spy-table [config]="{
  dataSource: { ... },
  columns: [ ... ],
  itemSelection: {
    enabled: true,
  },
  batchActions: {
    enabled: true,
    actions: [
      { id: 'edit', title: 'Edit', type: 'edit-action' },
      { id: 'update', title: 'Update', type: 'update-action' },
    ],
    noActionsMessage: 'No available actions for selected rows',
    rowIdPath: 'sku',

>     availableActionsPath: ['path.to.actions'],

Reviewer: ```suggestion
    availableActionsPath: 'path.to.actions',
```
